lib/src/git.rs

```diff
@@ -969,6 +969,8 @@ pub fn reset_head(
         };
         if !skip_reset {
             git_repo.reset(new_git_commit.as_object(), git2::ResetType::Mixed, None)?;
+            // Refresh the index since some scripts expect that.
+            git_repo.index()?.update_all(["."], None)?;
```

Comment:
Appears that this does the opposite thing. What we want to do here is to synchronize metadata if indexed entries are unchanged. However, `update_all()` calculates diff and stages new "changed" entries.

BTW, everything is done in-memory. We'll need to call `index.write()` at the end.